### PR TITLE
Request location permissions in demo app

### PR DESCRIPTION
**Pull request by bsudekum**
Otherwise, the modal disappears before you can accept.

/cc @1ec5 @frederoni @ericrwolfe

### CODE REVIEW

File: Examples/Swift/ViewController.swift [hunk 1]
```diff
@@ -33,13 +33,19 @@ class ViewController: UIViewController, MGLMapViewDelegate {
...
     // In this example, we show you how you can create custom UIView that is used to show the user's location.
     // Set `showCustomUserPuck` to true to view the custom user puck.
     var showCustomUserPuck = false
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        
+        locManager.delegate = self
+        locManager.requestAlwaysAuthorization()
```
Comment (ericrwolfe) [inline Examples/Swift/ViewController.swift]: Let's change to `whenInUse`. No reason to use `always` for turn-by-turn nav.

File: Examples/Swift/Base.lproj/Main.storyboard [hunk 2]
```diff
@@ -27,16 +27,6 @@
                                 <gestureRecognizers/>
                                 <userDefinedRuntimeAttributes>
                                     <userDefinedRuntimeAttribute type="string" keyPath="styleURL__" value="mapbox://styles/mapbox/navigation-preview-day-v2"/>
-                                    <userDefinedRuntimeAttribute type="number" keyPath="latitude">
-                                        <real key="value" value="0.0"/>
```
Comment (1ec5) [inline Examples/Swift/Base.lproj/Main.storyboard]: Removing these user-defined runtime attributes probably resolves an issue I’d been seeing where we start at Null Island at launch and quickly ease to the user’s location.

File: Examples/Swift/ViewController.swift [hunk 3]
```diff
@@ -33,13 +33,19 @@ class ViewController: UIViewController, MGLMapViewDelegate {
 
     var exampleMode: ExampleMode?
     
+    var locManager = CLLocationManager()
```
Comment (1ec5) [inline Examples/Swift/ViewController.swift]: Nit: spell out `locationManager`.

Comment (1ec5) [reply Examples/Swift/ViewController.swift]: Looks like there’s a conflict with a preexisting `locationManager()`. Can we rename that method?

File: Examples/Swift/Base.lproj/Main.storyboard (hunk 2 above)
Comment (bsudekum) [reply Examples/Swift/Base.lproj/Main.storyboard]: I have seen so many issues when this is added....

File: Examples/Swift/ViewController.swift [hunk 4]
```diff
@@ -94,7 +94,7 @@ class ViewController: UIViewController, MGLMapViewDelegate, CLLocationManagerDel
         
         let locationManager = ReplayLocationManager(locations: Array<CLLocation>.locations(from: filePath))
         
-        let navigationViewController = NavigationViewController(for: route, locationManager: locationManager)
+        let navigationViewController = NavigationViewController(for: route, locationManager: naviationLocationManager())
```
Comment (1ec5) [inline Examples/Swift/ViewController.swift]: Nit: naviation → navigation

**Comment by bsudekum** [comment]
@1ec5 this is ready for review.

**Pull request merged**